code/WorkInProgress/MechanicMadness.dm

```diff
@@ -15,6 +15,254 @@
+			src.icon_state=initial(src.icon_state)+"_closed"
+		return
+	proc/close_storage_menus() // look for nerds who still have the container UI open and close it for them
+		for (var/mob/chump in range(get_turf(src),1))
+			for(var/datum/hud/storage/hud in chump.huds)
+				if(hud.master==src) hud.close.clicked()
+		for (var/obj/storage/storage in range(get_turf(src),1)) // this amount of for() loops cannot be healthy, but oh well
+			for (var/mob/chump in storage.contents)
+				for(var/datum/hud/storage/hud in chump.huds)
+					if(hud.master==src) hud.close.clicked()
+	proc/destroy_outside_connections()
+		//called when the cabinet is unanchored
+		var/discons=0
+		for (var/atom/comp in src.contents)
+			if(!istype(comp.mechanics))
+				continue
+			for(var/atom/M in comp.mechanics.connected_outgoing)
+				if (M.loc==src)
+					continue
+				if(M.mechanics)
+					M.mechanics.connected_incoming-=comp
+				comp.mechanics.connected_outgoing-=M
+				discons++
+			for(var/atom/M in comp.mechanics.connected_incoming)
+				if (M.loc==src)
+					continue
+				if(M.mechanics)
+					M.mechanics.connected_outgoing-=comp
+				comp.mechanics.connected_incoming-=M
+				discons++
+		return discons
+	disposing()
+		..()
+		src.contents=null
+		return
+	MouseDrop(atom/target)
+		if(!istype(usr))
+			return
+		if(src.open && target == usr)
+			return ..()
+		if(!src.anchored && target != usr)
+			return ..()
+		return
+	get_desc()
+		.+="[src.welded ? " It is welded shut." : ""][src.open ? " Its cover has been opened." : ""][src.anchored ? "It is [src.open || src.welded ? "also" : ""] anchored to the ground." : ""]"
+	housing_large // chonker
+		can_be_welded=1
+		can_be_anchored=1
+		slots=23 // wew, dont use this in-hand or equipped!
+		name="Component Cabinet"
+		desc="A rather chunky cabinet for storing up to 23 active mechanic components at once.<br>It can only be connected to external components when bolted to the floor.<br>"
+		w_class = 4.0 //all the weight
+		density=1
+		anchored=0
+		icon_state="housing_cabinet"
+		flags = FPRINT | EXTRADELAY | CONDUCT
+		attack_hand(mob/user as mob)
+			if(src.loc==user)
+				src.set_loc(get_turf(src))
+				user.drop_item()
+				return
+			return MouseDrop(user)
+		attack_self(mob/user as mob)
+			src.set_loc(get_turf(user))
+			user.drop_item()
+			return
+		MouseDrop(atom/target)
+		// thanks, whoever hardcoded that pick-up action into obj/item/MouseDrop()!
+			if(istype(target,/obj/screen/hud))
+				return
+			if(target.loc!=get_turf(target) && !isturf(target)) //return if dragged onto an item in another object (i.e backpacks on players)
+				return // you used to be able to pick up cabinets by dragging them to your backpack
+			return ..()
+	housing_handheld
+		var/obj/item/mechanics/trigger/trigger/the_trigger
+		slots=7
+		name="Device Frame"
+		desc="A massively shrunken component cabinet fitted with a handle and an external button. Due to the average mechanic's low arm strength, it only holds 6 components."
```

Comment:
![image](https://user-images.githubusercontent.com/4741640/86546487-042e7680-beea-11ea-9c0a-befcbb0dadd9.png)

Reply:
The "7th" slot is taken up by the permeant button. So there's 6 slots to add/remove things from.

Reply:
comment should be added notating that then imo

```suggestion
		slots=7 // One slot used by the permanent button
		name="Device Frame"
		desc="A massively shrunken component cabinet fitted with a handle and an external button. Due to the average mechanic's low arm strength, it only holds 6 components."
```